modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java

```diff
@@ -0,0 +1,1788 @@
+/*
```

Comment:
Can we avoid adding yet another 2k-lines class? I'd be happy if this would be decomposed into 4-5 smaller ones. Some of the methods (like `coordinator`) could be moved to utility classes.

Reply:
The implementation is based on `GridServiceProcessor` but incompatible with it. Implementation has been detached to be able to use both implementations.
I'd like to avoid complete reworking of the class, in phase 1, to not complicate review.

Reply:
This kind of implies that reworking of the class is scheduled for phase 2 :) If that's the case then great!